code_search/kubeflow/components/nms.libsonnet

```diff
@@ -1,20 +1,19 @@
 local baseParams = std.extVar("__ksonnet/params").components["nmslib"];
 
 {
-
-  nmsContainer(params, env):: {
-    apiVersion: "extensions/v1beta1",
-    kind: "Deployment",
+  spec(params, env, apiVersion="extensions/v1beta1", kind="Deployment"):: {
```

Comment:
Why are you parameterizing the kind for this component? Why isn't nms always a Deployment or always a Job?

Reply:
The step before deploying the search server is the index creation. This is a one-off job. I observed that the index was being created multiple times over. This was because when using the kind "Deployment", restartPolicy can only be "Always" (which was kind of expected in hindsight) and even if the container exited, it just restarted building the index again. "Job" allows restartPolicy of "onFailure" which made sense in this case.

Reply:
I'm still confused. Lets chat about this.